### ISSUE TITLE
token refresh

### ISSUE THREAD

**Issue by sooslaca** [JoDehli/PyLoxone#153]
Hi @JoDehli ,

Nice job of yours indeed, without knowing much of Loxone API, I'm not sure I get how token refresh is working.
https://github.com/JoDehli/PyLoxone/blob/master/custom_components/loxone/api.py#L287
You set "valid_until"  (set_vaild_until) but I cannot see where the actual token is being set.

Is there not a `self._token.set_token()` missing somewhere in the `_refresh_token()` function?

Thanks

**Comment by JoDehli** [comment]
I did not check it in detail but I think you are right. As it is already a old implementation of the pyloxone api I will not change it. I am currently working on a complete rewrite of the api which does not use the websocket lib anymore. It uses the aiohttp lib. In the future I want to separate the communication between the miniserver and the home assistant part completely. 

I am really not happy with the implementation as it is today. I started it as a project for myself but no I work most of the time to implement things for other people and I have no time to work on the implementation itself. 

Do have python skills and want to support me with the programming?
